CONTRIBUTING.md

```diff
@@ -0,0 +1,133 @@
+
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
+contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+## What should I know before I get started?
+
+The DataOps repo is a collection of end-to-end samples, smaller single technology samples, and utility code snippets which aims help data engineers operationalize their data systems using DevOps principals with Microsoft technologies. Each sample showcases one or more of the following engineering fundamental pillars: **Infrastructure-as-code (IaC)**, **CI/CD**, **automated testing**, and **observability** but may also cover additional principals as covered by the [Microsoft Azure Well-Architected Framework](https://docs.microsoft.com/en-us/azure/architecture/framework/).
+
+## Creating a new sample
+
+### Pre-requisites
+
+**First**, look through existing samples to ensure you are not duplicating work. Also, look through Github issues if there is any similar work in progress (ie. issues with PRs associated to them).
+
+**Second**, once you've confirmed this, you need to decide whether you are creating either an End-to-End Sample or a Single Technology Sample.
+
+1. [End-to-End Samples](/e2e_samples/) - Reference-architecture style DataOps samples typically utilizing a number of Microsoft services and technologies.
+   - **Purpose**: Showcase a number of DataOps key concepts in a concrete, end-to-end data pipeline.
+
+2. [Single Technology Samples](/single_tech_samples/) - Smaller, modular samples focusing on a single Azure service showcasing one or two DataOps key concepts.
+   - **Purpose**: allows Developers and Data Engineers to easily choose and reuse individual samples depending on their requirements without the complexity of an end-to-end sample. Being small and modular also allows for showcasing different implementations for the same concept.
+
+3. [Utilities](/utilities/) - Reusable, standalone scripts that help with a specific operational task.
+   - **Purpose**: Easily find helpful scripts used in common tasks to manage a data engineering system.
+
+**Third**, decide which [key concepts](https://docs.microsoft.com/en-us/azure/architecture/framework/) are you showcasing in your sample. See [Microsoft Azure Well-Architected Framework](https://docs.microsoft.com/en-us/azure/architecture/framework/) for additional information. If you've decided to build a Single Technology Sample, decide which technology will be showcasing the key concepts.
+
+**Fourth**, locate where in the repository structure your sample will sit. Decide on the name of the sample -- short descriptive names are best. It is recommended to **not** choose a name too lengthy and complicated as it will be used across the repo (folder name, script paths, deep-linking in docs, labels, etc.).
```

Comment:
```suggestion
"**Fourth**, determine the appropriate location for your sample within the repository structure. Select a short, descriptive name for the sample, avoiding lengthy or complex names, as it will be used throughout the repo (e.g., folder names, script paths, documentation links, and labels)."
```